chatmaild/src/chatmaild/metadata.py

```diff
@@ -117,7 +118,30 @@ def handle_dovecot_request(msg, transactions, notifier, metadata, iroh_relay=Non
         if keyname[0] == "priv" and keyname[2] == metadata.DEVICETOKEN_KEY:
             metadata.add_token_to_addr(addr, value)
         elif keyname[0] == "priv" and keyname[2] == "messagenew":
-            notifier.new_message_for_addr(addr, metadata)
+            try:
+                proc = subprocess.run(
+                    [
+                        "doveadm",
+                        "fetch",
+                        "-u",
+                        addr,
+                        "--",
+                        "hdr.auto-submitted",
+                        "uid",
+                        value,
+                    ],
+                    capture_output=True,
+                    check=True,
+                )
+                notify = proc.stdout not in [
+                    b"hdr.auto-submitted: auto-generated\n",
+                    b"hdr.auto-submitted: auto-replied\n",
+                ]
+            except FileNotFoundError:
+                # doveadm not available, don't notify
```

Comment:
better do something like `doveadm_path = subprocess.check_output(["which", "doveadm"], text=True).strip()` at module level and use that.